src/components/search/outsideAlerter.js

```diff
@@ -0,0 +1,18 @@
+import { useRef, useState, useEffect } from "react";
+
+export const useOutsideAlerter = (initialValue : boolean) => {
+    const ref = useRef<HTMLDivElement>(null)
+    const [ visible, setVisible ] = useState(initialValue)
+    const handleClickOutside = (e) => {
+        if(ref.current && !ref.current.contains(e.target)) setVisible(false)
+    }
+
+    useEffect(() => {
+        document.addEventListener("click", handleClickOutside, true)
+        return () => {
+            document.removeEventListener("click", handleClickOutside, true)
+        }
+    },[ref])
+
+    return {visible, setVisible, ref }
+}
```

Comment:
I see what you're going for here, seems cool, though definitely clean up that vestigial typescript. It doesn't appear to be used anywhere in the code yet, can you confirm that it's working correctly?

Reply:
I'll have to come back to this I had it plugged in somewhere , but I can't remember now. I removed this file to try to move forward and will tackled the outside-alerter as a later action item.